bookmarks.go

```diff
@@ -0,0 +1,159 @@
+package slack
+
+import (
+	"context"
+	"net/url"
+)
+
+type Bookmark struct {
+	ID        string   `json:"id"`
+	ChannelID string   `json:"channel_id"`
+	Title     string   `json:"title"`
+	Link      string   `json:"link"`
+	Emoji     string   `json:"emoji"`
+	IconURL   string   `json:"icon_url"`
+	Type      string   `json:"type"`
+	Created   JSONTime `json:"date_created"`
+	Updated   JSONTime `json:"date_updated"`
+	Rank      string   `json:"rank"`
+
+	LastUpdatedByUserID string `json:"last_updated_by_user_id"`
+	LastUpdatedByTeamID string `json:"last_updated_by_team_id"`
+
+	ShortcutID string `json:"shortcut_id"`
+	EntityID   string `json:"entity_id"`
+	AppID      string `json:"app_id"`
+}
+
+type AddBookmarkParameters struct {
+	Title    string // A required title for the bookmark
+	Type     string // A required type for the bookmark
+	Link     string // URL required for type:link
+	Emoji    string // An optional emoji
+	EntityID string
+	ParentID string
+}
+
+type EditBookmarkParameters struct {
+	Title *string // Change the title. Set to "" to clear
+	Emoji *string // Change the emoji. Set to "" to clear
```

Comment:
Done to capture the difference between not modifying a value and setting it to empty since `""` is used in the API to clear the values. Happy to hear a better idea on how to do optionals here.

Reply:
@winston-stripe 
I saw API doc, nodejs and python sdk.
That's definitely optional.

- https://github.com/slackapi/node-slack-sdk/blob/%40slack/web-api%406.7.0/packages/web-api/src/methods.ts#L1252-L1254
- https://github.com/slackapi/python-slack-sdk/blob/v3.15.2/slack_sdk/web/client.py#L1588-L1590

But I can't find clear if this field put empty. do you have any documents URL?

Reply:
@winston-stripe 
I saw API doc, nodejs and python sdk.
That's definitely optional.

- https://github.com/slackapi/node-slack-sdk/blob/%40slack/web-api%406.7.0/packages/web-api/src/methods.ts#L1252-L1254
- https://github.com/slackapi/python-slack-sdk/blob/v3.15.2/slack_sdk/web/client.py#L1588-L1590

But I can't find clear if this field put empty. do you have any documents URL?

Reply:
My understanding comes from testing in the slack API tester where a value not present (i.e. not included) results in no change while the `emoji=` for param encoding (ostensibly equivalent to `{"emoji": ""}`) results in resetting the value to empty.

https://user-images.githubusercontent.com/4950148/159413842-03a72c22-1b11-4fb2-8eb0-eff0c3518a24.mp4

Reply:
I just added ran a quick test case using the `EditBookmark()` call in Go and it behaved as expected. Linked is code I ran and output.

https://gist.github.com/R167/9347a9175247bc86a41ef1612413d42a

Reply:
@R167 Thanks! I'll try to run your gist.

Reply:
@winston-stripe Ah, sorry. Please add GoDoc these fields.

Reply:
@kanata2 can you clarify, sorry. Looks like they are godoc'd? or was there a specific bit you wanted me to add? 🙂

Reply:
Oh, sorry. I was looking at another part. No problem.